packages/gatsby/index.d.ts

```diff
@@ -52,13 +52,13 @@ export const prefetchPathname: (path: string) => void
  * export default (props: IndexProps) => {
  *   ..
  */
-export type PageProps<DataType = object, PageContextType = object> = {
+export type PageProps<DataType = object, PageContextType = object, LocationState = object> = {
```

Comment:
```suggestion
export type PageProps<DataType = object, PageContextType = object, LocationState =  = WindowLocation["state"]> = {
```

Rather than defaulting to `object`, it would be better to default to WindowLocation's own default type. (That is, admittedly, currently aliased from `any`, but that may change)

Reply:
I think the current alias is `{} | null | undefined` ("`PoorMansUnknown`"), taken from [`History.LocationState`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/history/index.d.ts#L55). 
But let's use it anyway.